.github/workflows/acceptance_tests_gke.yaml

```diff
@@ -52,49 +59,145 @@ jobs:
         uses: "google-github-actions/auth@v0"
         with:
           credentials_json: ${{ secrets.GOOGLE_CREDENTIALS }}
-          access_token_lifetime: '10800s'
+          access_token_lifetime: "10800s"
       - name: Install Terraform
-        uses: hashicorp/setup-terraform@v1
+        uses: hashicorp/setup-terraform@v2
         with:
           terraform_version: ${{ github.event.inputs.terraformVersion }}
           terraform_wrapper: false
-      - name: "Set up Cloud SDK"
-        uses: "google-github-actions/setup-gcloud@v0"
-        with:
-          install_components: "beta"
-      - name: "Initialize gcloud SDK"
-        run: "gcloud init"
       - name: Provision GKE Cluster
-        working-directory: ./kubernetes/test-infra/gke
+        working-directory: ${{ github.workspace }}/kubernetes/test-infra/gke
         env:
           TF_VAR_kubernetes_version: ${{ github.event.inputs.kubernetesVersion }}
           TF_VAR_workers_count: ${{ github.event.inputs.workersCount }}
           TF_VAR_node_machine_type: ${{ github.event.inputs.nodeMachineType }}
         run: |
-          env
           terraform init
           terraform apply -auto-approve
-          export CLUSTER_NAME=`terraform output -raw cluster_name`
-          echo "GOOGLE_CLUSTER_NAME=$CLUSTER_NAME" >> $GITHUB_ENV
-          gcloud beta container clusters update --enable-service-externalips $CLUSTER_NAME --zone $GOOGLE_ZONE
-      - name: Get GKE Credentials
-        env:
-          KUBECONFIG: ${{ github.workspace }}/kubernetes/test-infra/gke/kubeconfig
-        uses: 'google-github-actions/get-gke-credentials@v0'
-        with:
-          cluster_name: ${{ env.GOOGLE_CLUSTER_NAME }}
-          location: ${{github.event.inputs.zone}}
-          project_id: ${{ secrets.GOOGLE_PROJECT }}
-          use_auth_provider: true
-      - name: Run Acceptance Test Suite
+      - name: "Persist state"
+        uses: actions/upload-artifact@v3
+        with:
+          name: gke-cluster
+          retention-days: 1
+          path: |
+            ${{ github.workspace }}/kubernetes/test-infra/gke/kubeconfig
+            ${{ github.workspace }}/kubernetes/test-infra/gke/terraform.tfstate
+
+  generate-case-matrix:
+    name: "Generate matrix of test cases"
+    runs-on: ubuntu-latest
+    outputs:
+      test-case-matrix: ${{ steps.generate.outputs.test-case-matrix }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Read go-version
+        id: go-version
+        uses: juliangruber/read-file-action@v1
+        with:
+          path: ./.go-version
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ steps.go-version.outputs.content }}
+      - name: "Generate testcase matrix"
+        id: generate
+        run: |
+          make test-compile
+          ./kubernetes.test -test.list '${{ github.event.inputs.runTests }}' | go run scripts/batchacc.go -sort -depth 3 | tee groups.json
```

Comment:
I think this thing might be a candidate for it's own github action – `go-test-parallelizer` 😎

Reply:
Yeah, was thinking the same. This should be a re-usable action as well as the one running the actual tests. Then we can use the same logic in all cloud specific environments.

I'd proposed to do that once we get to extending this approach to the next cloud since it's a bit finicky to extract and this work has already gone overboard in duration.